addons/plugins/contracts_payment_plugin.py

```diff
@@ -93,31 +145,39 @@ def process_payment(
         # adjust the balance
         # return status?
         token = payment_information.token
-        try:
-            ContractsRepository.charge_contract(
-                token, billed_amount=payment_information.amount
+        contract = self.CONTRACTS_REPOSITORY.get_contract_by_id(token)
+        if not self.CONTRACTS_REPOSITORY.user_has_contract(
+            payment_information.customer_email, token
+        ):
+            return GatewayResponse(
+                is_success=False,
+                action_required=False,
+                kind="capture",
```

Comment:
Saleor has predefined constants for transaction kinds, in this case: `TransactionKind.CAPTURE`. It's a minor thing, but it's always safer to use them than hardcoded strings